presentation/src/main/java/me/androidbox/presentation/event/viewmodel/EventViewModel.kt

```diff
@@ -206,6 +207,32 @@ class EventViewModel @Inject constructor(
         }
     }
 
+    fun fetchEventById(eventId: String) {
+        viewModelScope.launch {
+            eventRepository.getEventById(eventId).collectLatest { responseState ->
+                when(responseState) {
+                    ResponseState.Loading -> {
+                        /** TODO Show loading */
+                    }
+                    is ResponseState.Success -> {
+                        /* Update the state */
+                        val event = responseState.data
+                        _eventScreenState.update { eventScreenState ->
+                            eventScreenState.copy(
+                                eventId = event.id,
+                                eventTitle = event.title,
+                                eventDescription = event.description,
+                            )
+                        }
+                    }
+                    is ResponseState.Failure -> {
+                        Log.d("EVENT_DETAIL", "${responseState.error.message}")
+                    }
+                }
+            }
+        }
```

Comment:
Get the event from the DB and updates the state. I am just testing with the title and description at the moment.